Modernizace CSS

eksyska opened the pull request:
- Zjednodušeny názvy css tříd, aby nebyly tak složité a šly lépe kombinovat
- Elegantnější header, díky kterému je efektivněji využitý prostor
- Sjednocení tlačítek v menu (nyní ve formě ikon)
- Doplnění zprávy pro demo verzi, že s vlastním účtem lze zakládat vlastní třídy
- Kvůli aktualizaci sass kompilátoru se výrazněji přepsaly css soubory

eksyska committed:
Přepis tříd typu "data-item"

eksyska committed:
Přepis tříd uvnitř položek data-item

eksyska committed:
Modernější layout

Zatím zejména pro desktop

eksyska committed:
Plynulejší menu

Při volbě části nyní tlačítka nenahradí text, ale vysunou se.
Díky této změně také bude jednodušší přidat tlačítko na plánovanou možnost zkoušení z vygenerovaného testu.

eksyska committed:
Drobné úpravy pohledů v sekci study

Zmenšeno písmo v labelu za účelem sjednocení, zmenšení obsahu v learn.phtml, aby se nezobrazoval scrollbar

eksyska committed:
Oprava velikosti info tlačítka

eksyska committed:
Přepis tříd pro pohled naturals.phtml

eksyska committed:
Oprava názvu obrázku

eksyska committed:
Přepis templatů pro data-items

eksyska committed:
Přepis tříd v edit.phtml

eksyska committed:
Drobné změny

Úprava scrollbaru, nastavení velikosti labelu

eksyska committed:
Vylepšení pohledu menu.phtml

nový jednodušší a hezčí header (i pro mobily)
zmenšení fontu pro mobily

eksyska committed:
Sjednocení tlačítek, oprava javascriptu

Tlačítka v data-item.action mají nyní všechna formu ikony.
Navíc byl opraven javascript naturals.phtml, který kvůli přepisu tříd nefungoval tak, jak měl.

eksyska committed:
Oprava názvů tříd pro pohled naturals.phtml

eksyska committed:
Přepis tříd v administrativních pohledech a reportsTableManage.phtml

eksyska committed:
Oprava barvy textu ve footeru

eksyska committed:
Doplnění zprávy do menu demo účtu

Informuje uživatele, že s vlastním účtem mohou sami zakládat třídy a poznávčaky

eksyska committed:
Oprava gramatiky

ShadyMedic commented on Models/ChangelogManager.php:
```diff
@@ -48,6 +48,11 @@ private function loadChangelog(): void
         curl_setopt($curl, CURLOPT_URL, self::GITHUB_API_RELEASES_URL.self::RELEASE_IDS[self::LATEST_VERSION]);
         curl_setopt($curl, CURLOPT_RETURNTRANSFER, true);
+
+        // fix chyby s SSL certifikátem - pouze pro vývoj stránek
```
TODO Přidat do index.php nějakou konstantu `PRODUCTION_MODE`, která toto nastavení povolí pouze při nastavení na `false`. To samé i pro věci kolem kontroly SSL a e-mailů.

ShadyMedic replied:
Spraveno, viz #272

ShadyMedic commented on Views/head.phtml:
```diff
@@ -38,9 +38,9 @@
-                    <a id="help" class="btn inline white" target='_blank' rel="noopener" href="https://github.com/HonzaSTECH/Poznavacky/wiki">Nápověda</a>
-                    <a id="issues" class="btn inline white" onclick="showLogin()" rel="noopener" target='_blank' href="https://forms.gle/sreDqmdtNT4Z3jwr5">Nalezl*a jsi problém?</a>
-                    <a id="terms" class="btn inline white" target='_blank' rel="noopener" href='/legal'>Právní informace</a>
+                    <a id="help" class="btn inline" target='_blank' rel="noopener" href="https://github.com/HonzaSTECH/Poznavacky/wiki">Nápověda</a>
```
TODO: Aktualizovat jméno majitele repozitáře na ShadyMedic

ShadyMedic replied:
Spraveno v dc90c7de271c59db2544ecd127e6659777a7fa5e

ShadyMedic committed:
Doplnění přípony do "src" atributy pro ikonu

ShadyMedic committed:
Přidání statické třídy se systémovými nasaveními

Tato třída obsahuje aktuálně 2 nastavení – zda má aplikace běžet v produkčním nebo vývojovém módu a e-mailová schránka, do které mají chodit všechny e-maily, pokud běží ve vývojovém módu.
Ve vývojovém módu není vynucováno SSL připojení a e-maily vždy odchází jenom na sběrnou e-mailovou adresu.
V budoucnu budou nejspíše přidána další nastavení a využití vývojového módu.

ShadyMedic committed:
Přesun konstant z Db do Settings

Konstanty obsahovaly defaultní přihlašovací údaje k databázi – jakožto nastavení tak byly přesunuty

ShadyMedic committed:
Přesun konstant z EmailSender do Settings

Konstanty obsahovaly přihlašovací údaje k e-mailové schránce pro odesílání e-mailů – jakožto nastavení tak byly přesunuty

ShadyMedic committed:
Přesun konstanty z ChangelogManager do Settings

Konstanta obsahovala údaj o verzi aplikace

ShadyMedic committed:
Přesun konstanty z TestPicturesController do Settings

Konstanta obsahovala údaj o počtu obrázků pro testovací stránky odeslaných najednou

ShadyMedic committed:
Přesun konstanty z AnswerChecker do Settings

Konstanta obsahovala údaj o přísnosti hodnocení odpovědí na testové otázky

ShadyMedic committed:
Přesun konstanty z Logger do Settingsb

Konstanta obsahovala údaj o e-mailové adrese pro hlášení chy

ShadyMedic committed:
Přesun konstanty z Invitation do Settings

Konstanta obsahovala údaj o době platnosti pozvánky

ShadyMedic committed:
Přesun konstanty z LoginUser do Settings

Konstanta obsahovala údaj o době platnosti cookie pro okamžité přihlášení

ShadyMedic committed:
Přesun konstanty z RecoverPassword do Settings

Konstanta obsahovala údaj o době platnosti kódu pro obnovu hesla

ShadyMedic committed:
Přesun konstanty z PictureAdder do Settings

Konstanta obsahovala údaj o povolených typech souborů pro přidávání jako obrázky přírodnin

ShadyMedic committed:
Nahrazení všech výskytů starého jména majitele GitHub repozitáře

HonzaSTECH --> ShadyMedic

eksyska committed:
Oprava bugu, kvůli němuž nefungovalo opuštění třídy

eksyska committed:
Přepis tříd, změny v css souborech kvůli aktualizaci sass kompilátoru

eksyska committed:
Přepis tříd pro manage.phtml a tests.phtml

ShadyMedic committed:
Merge pull request #272 from ShadyMedic/environment-settings

Přesun mnoha konstant do nové třídy pro nastavení

ShadyMedic reviewed (approved):
LGTM
